### PR TITLE
Sockets cleanup

### PR DESCRIPTION
**Pull request by hovi**
- SpaceEngineers tests are closing connection correctly after each run.
- SocketEnvironment closing connection is safer.
- Timeouts for connecting and idle connection in SocketEnvironment.

Could be refactored better (especially tests), but if we move to kotlin, I don't think it's worth the effort.

### CODE REVIEW

File: src/main/java/spaceEngineers/environments/SocketEnvironment.java [hunk 1]
```diff
@@ -32,6 +33,8 @@
     SocketEnvironment(String host, int port) {
 
         int maxWaitTime = 20000;
+        int socketConnectionTimeout = 4000;
+        int socketDataTimeout = 4000;
```
Comment (PremekPaska) [inline src/main/java/spaceEngineers/environments/SocketEnvironment.java]: It's not critical, but I like when physical quantities have units in the name of the variable, such as:

maxWaitTimeMs
socketConnectionTimeoutMs
etc.

Comment (PremekPaska) [reply src/main/java/spaceEngineers/environments/SocketEnvironment.java]: Just a Java newbie question: why not add something like final / const?

**Review by PremekPaska — APPROVED** [review]
Awesome :)

I added one suggestion and one question.

File: src/main/java/spaceEngineers/environments/SocketEnvironment.java (hunk 1 above)
Comment (hovi) [reply src/main/java/spaceEngineers/environments/SocketEnvironment.java]: I renamed variables to match this.

Yes good point, final should be in 99% places of the java code, including method parameters. There's even settings in IDEA (or Eclipse), that adds it all automatically on save. I used to do that before switching to kotlin.
These variables don't make sense at all and they should be (final) constructor parameters with some sane defaults. imho :-)
In kotlin, 99% variables should be "val", which serves the same purpose.

**Review by PremekPaska — APPROVED** [review]
OK, so please combine all 3 commits that are related to opening connection / timeouts into one commit and merge it.

**Comment by PremekPaska** [comment]
Merged manually (fast forward, since it's only one commit).

**Pull request closed without merging**